netket/nn/symmetric_linear.py

```diff
@@ -160,19 +155,41 @@ def __call__(self, x: Array) -> Array:
         if self.mask is not None:
             kernel = kernel * jnp.expand_dims(self.mask, (0, 1))
 
-        kernel = self.full_kernel(kernel).reshape(-1, self.features, self.n_symm)
+        # Converts the convolutional kernel of shape (self.features, in_features, n_sites)
+        # to a full dense kernel of shape (self.features, in_features, n_symm, n_sites).
+        # result[out, in, g, r] == kernel[out, in, g^{-1}r]
+        kernel = jnp.take(kernel, jnp.asarray(self.symmetries), 2)
         kernel = jnp.asarray(kernel, dtype)
 
+        # x is      (batches,       in_featuers,         n_sites)
+        # kernel is (self.features, in_features, n_symm, n_sites)
         x = lax.dot_general(
             x,
             kernel,
-            (((x.ndim - 1,), (0,)), ((), ())),
+            (
+                (
+                    (
+                        x.ndim - 2,
+                        x.ndim - 1,
```

Comment:
```suggestion
                        1,2
```
This line is plain unreadable because of black. Do we care to support more than one batch dimensions???

Reply:
This should not be formatted like this in any case. If we need to keep `x.ndim`, I'd extract this parameter to a local variable, which hopefully would keep this expression to one line in the assignment.

Reply:
`x.ndim` is always 3 at this point unless you call it with more than one batch dimensions. I didn't check if that would break by this point, but it probably does somewhere...

Reply:
Note to self: ask JAX people to support negative axis labels in `dot_general`

Reply:
Also turns out that it fits neatly on one line, black just had a brain freeze

Reply:
> ask JAX people to support negative axis labels in dot_general 
I think one of the `log4j` exploits that is sending the internet bonkers those days is exactly because they support negative indices...

Reply:
"support" in the sense C allows you to index before or after an array?